IllegalArgumentException

Thank you very much with your help so far in #71!
After the refactoring I am a bit stuck. I currently try to run the current `master`. However, I am unable to do this after the refactoring.
The build runs normally and succesfully, unfortunately on starting the application it crashes. I am running in Android 5.1.
Before building, I have checked out master, then have done `git submodule update` and run `./pre-build.sh`.
I am not quite sure where this error does come from. Please find the crashing stack-trace below.


```
E/AndroidRuntime: FATAL EXCEPTION: main
                  Process: de.grobox.liberario.debug, PID: 23813
                  java.lang.RuntimeException: Unable to resume activity {de.grobox.liberario.debug/de.grobox.liberario.activities.MainActivity}: java.lang.IllegalArgumentException: type ANY cannot have ID
                      at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2986)
                      at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3017)
                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2392)
                      at android.app.ActivityThread.access$800(ActivityThread.java:151)
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
                      at android.os.Handler.dispatchMessage(Handler.java:102)
                      at android.os.Looper.loop(Looper.java:135)
                      at android.app.ActivityThread.main(ActivityThread.java:5254)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at java.lang.reflect.Method.invoke(Method.java:372)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:699)
                   Caused by: java.lang.IllegalArgumentException: type ANY cannot have ID
                      at com.google.common.base.Preconditions.checkArgument(Preconditions.java:122)
                      at de.schildbach.pte.dto.Location.<init>(Location.java:60)
                      at de.schildbach.pte.dto.Location.<init>(Location.java:0)
                      at de.grobox.liberario.data.RecentsDB.getLocation(RecentsDB.java:563)
                      at de.grobox.liberario.data.RecentsDB.getLocation(RecentsDB.java:551)
                      at de.grobox.liberario.data.RecentsDB.getFavLocationList(RecentsDB.java:64)
                      at de.grobox.liberario.data.RecentsDB.getFavLocationList(RecentsDB.java:76)
                      at de.grobox.liberario.adapters.LocationAdapter.getDefaultLocations(LocationAdapter.java:244)
                      at de.grobox.liberario.adapters.LocationAdapter.setDefaultLocations(LocationAdapter.java:219)
                      at de.grobox.liberario.ui.LocationView.reset(LocationView.java:372)
                      at de.grobox.liberario.ui.LocationView.resetIfEmpty(LocationView.java:378)
                      at de.grobox.liberario.fragments.DirectionsFragment.refreshAutocomplete(DirectionsFragment.java:476)
                      at de.grobox.liberario.fragments.DirectionsFragment.onResume(DirectionsFragment.java:232)
                      at android.support.v4.app.Fragment.performResume(Fragment.java:2126)
                      at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1151)
                      at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1290)
                      at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1272)
                      at android.support.v4.app.FragmentManagerImpl.dispatchResume(FragmentManager.java:2159)
                      at android.support.v4.app.FragmentController.dispatchResume(FragmentController.java:223)
                      at android.support.v4.app.FragmentActivity.onResumeFragments(FragmentActivity.java:507)
                      at android.support.v4.app.FragmentActivity.onPostResume(FragmentActivity.java:496)
                      at android.support.v7.app.AppCompatActivity.onPostResume(AppCompatActivity.java:172)
                      at android.app.Activity.performResume(Activity.java:6090)
                      at android.app.ActivityThread.performResumeActivity(ActivityThread.java:2975)
                      at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3017) 
                      at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2392) 
                      at android.app.ActivityThread.access$800(ActivityThread.java:151) 
                      at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303) 
                      at android.os.Handler.dispatchMessage(Handler.java:102) 
                      at android.os.Looper.loop(Looper.java:135) 
                      at android.app.ActivityThread.main(ActivityThread.java:5254) 
                      at java.lang.reflect.Method.invoke(Native Method) 
                      at java.lang.reflect.Method.invoke(Method.java:372) 
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:904) 
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:699)
```

--- a/src/de/grobox/liberario/data/RecentsDB.java
+++ b/src/de/grobox/liberario/data/RecentsDB.java
@@ -96,8 +96,8 @@ public class RecentsDB {
 	}
 
 	public static void updateFavLocation(Context context, Location loc, FavLocation.LOC_TYPE loc_type) {
-		if(loc == null || loc.type == LocationType.COORD) {
-			// don't store GPS locations
+		if(loc == null || loc.type == LocationType.COORD || loc.type == LocationType.ANY) {
+			// don't store GPS or ANY locations
 			return;
 		}
 

